samples/AntiforgerySample/Startup.cs

```diff
@@ -10,13 +23,55 @@ public class Startup
     {
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddAntiforgery();
+            services.AddRouting();
+            
+            // Angular's default header name for sending the XSRF token.
+            services.AddAntiforgery(options => options.HeaderName = "X-XSRF-TOKEN");
+
+            services.AddSingleton<TodoRepository>();
         }
 
-        public void Configure(IApplicationBuilder app)
+        public void Configure(IApplicationBuilder app, IAntiforgery antiforgery, IOptions<AntiforgeryOptions> options, TodoRepository repository)
         {
+            app.Use(next => context =>
+            {
+                if (context.Request.Path == "/")
+                {
+                    // We can send the request token as a JavaScript-readable cookie, and Angular will use it by default.
+                    var tokens = antiforgery.GetAndStoreTokens(context);
+                    context.Response.Cookies.Append("XSRF-TOKEN", tokens.RequestToken, new CookieOptions() { HttpOnly = false });
```

Comment:
It would be nice to find a way to build this in as well. We could make this a standalone middleware if necessary. Right now it's up to the app developer to know when to set the cookie.